### DISCUSSION TITLE
Optional GUI Proposal: Intuitive Flow Chart Visualization for Fan Control Configurations

### DISCUSSION THREAD

**Discussion by Hyporeal** [Rem0o/FanControl.Releases#2424] [Ideas]
ISSUE: The current GUI works, but can present challenges with managing complex systems, often requiring repetitive referencing of sensors and fan objects.

PROPOSAL: Introduce a 'Flow Chart Mode' that simplifies the process of configuring fan and sensor interactions. This visual mapping tool would allow users to more easily create connections, significantly lowering the learning curve for new users and enhancing user-friendliness. It streamlines setup, eliminates repetitive entries, and supports easier modifications, making the management of complex systems both intuitive and efficient.

_Standard Mode (Below)_
![Standard Mode](https://github.com/Rem0o/FanControl.Releases/assets/87534492/8e470675-291c-4a91-9aec-5cc43853564f)

_Flow Chart Mode (Below)_
![Flow Chart Mode](https://github.com/Rem0o/FanControl.Releases/assets/87534492/b056e81f-4935-47b0-8472-7bda333f83ee)

**Comment by Rem0o** [comment]
It's a cool idea, but would kind of be a nightmare to code with the various edge cases, and finding the appropriate x/y position to avoid a spaghetti mess with connections.  

Without jumping to a technical solution right away, if we go back a little, would you say the "problem" is: 

> As configurations get more complex, there is a lack of flexibility and visual cues to organize the different relations between the different component of your config?

That kinda matches your issue. So before going into this quite complex solution, I would like to know if more simple solutions would offer great improvement. 

For eg:

1) Tiles connection highlights: When you hover your mouse over a tile, it's "connected" tiles will highlight. 
2) Free form home screen: Allow to move tiles of any types where-ever.

**Reply by Hyporeal** [reply]
Firstly, I want to say that this project is fantastic and I'm in no way complaining. I was able to get the configuration to work, but I do have those near to me that would benefit from a visual representation.

>1. Tiles connection highlights: When you hover your mouse over a tile, it's "connected" tiles will highlight.
>2. Free form home screen: Allow to move tiles of any types where-ever.

**That sounds like a fantastic addition! I showed it to a primarily visual person and the idea seem to really help them.** 

Perhaps, inspiration could be drawn from various example flow charts, mind maps, and mind mapping tools for future incremental development.

**Comment by ionGL** [comment]
When there are lots of controls and curves, it can get hard to follow what's going on.

Here is my setup. 10 controls, 19 curves.

![fancotnrolmess](https://github.com/Rem0o/FanControl.Releases/assets/80151979/24a2b37f-1b7a-489f-8ee9-d8a230e989d0)

The same setup in argus. 8 "regular" fan controls.

![argus](https://github.com/Rem0o/FanControl.Releases/assets/80151979/6ffb081a-5ea6-4eec-904f-26fd38a3d6f6)

Gpu tab.

![argusgpu](https://github.com/Rem0o/FanControl.Releases/assets/80151979/06d19cc3-7433-473f-ae48-bb8aaa30a47b)

Curve manager.

![arguscurvemanager](https://github.com/Rem0o/FanControl.Releases/assets/80151979/f2658ec1-76ba-46de-a207-b9c1880b7636)

I don't know how this comparation helps, but it's worth a look.

**Reply by Hyporeal** [reply]
That's very thoughtful of you to share!
